muxer/muxer.go

```diff
@@ -57,9 +58,10 @@ type Muxer struct {
 }
 
 // New creates a new Muxer object and starts the read loop
-func New(conn net.Conn) *Muxer {
+func New(connp *net.Conn) *Muxer {
 	m := &Muxer{
-		conn:              conn,
+		conn:              *connp,
+		connp:             connp,
```

Comment:
I don't really understand what you're trying to do here with 2 copies of the net.Conn object, one of which is a pointer. If the point is just to see if we've been provided an actual `net.Conn` object at some point, there are probably "better" ways to do that.

Reply:
the idea here is just to set the conn to nil so to prevent the Ouroboros.Close to double close it, probably the best way is just remove the net.Conn Close in the Ouroboros.Close as the "network thing" is a muxer responsability.
Basically I did this ugly thing to avoid to remove [this](https://github.com/blinklabs-io/gouroboros/blob/ea94cd02e649dc22e2f0dd957c05149dfbbcaf2a/ouroboros.go#L151)